From dde180a54f64a7db523d21ac594fc9c7a04cfbbe Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Fri, 5 Oct 2018 17:31:20 +0000
Subject: [PATCH] TRITON-844 booter mangles dhcpd option 82 Reviewed by: Cody
 Peter Mello <melloc@writev.io> Approved by: Cody Peter Mello
 <melloc@writev.io>

---
 lib/dhcpd.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/dhcpd.js b/lib/dhcpd.js
index 003cefc..63052f9 100755
--- a/lib/dhcpd.js
+++ b/lib/dhcpd.js
@@ -154,11 +154,11 @@ DHCPD.prototype.handleMessage = function (msg, peer) {
      *
      * [<option id>, <option length>, <variable length option data>]
      */
-    if (packet.options[82]) {
+    if (packet.options[82] && Array.isArray(packet.options[82])) {
         log.info({option_82: packet.options[82]}, 'Found DHCP Option 82, '
             + 'getting sub-options');
 
-        var relayopt = packet.options[82];
+        var relayopt = packet.options[82].slice();
         var suboptlen;
         var suboptid;
         var subopt;
-- 
2.21.0

